core/images_test.go

```diff
@@ -27,3 +37,105 @@ var (
 )
+
+func TestOpenBazaarNode_maybeMigrateImageHashes(t *testing.T) {
+
+	ipfsNode, err := coremock.NewMockNode()
+	if err != nil {
+		t.Fatal(err)
+	}
+	testRepo, err := schema.NewCustomSchemaManager(schema.SchemaContext{
+		DataPath:        schema.GenerateTempPath(),
+		TestModeEnabled: true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = testRepo.BuildSchemaDirectories(); err != nil {
+		t.Fatal(err)
+	}
+	defer testRepo.DestroySchemaDirectories()
+
+	node := OpenBazaarNode{IpfsNode: ipfsNode, RepoPath: testRepo.DataPath()}
+
+	listing := factory.NewListing("testListing")
+	for _, image := range listing.Item.Images {
+		err := ioutil.WriteFile(path.Join(testRepo.DataPath(), "root", "images", "large", image.Filename), []byte("image"), os.ModePerm)
+		if err != nil {
+			t.Error(err)
+		}
+		err = ioutil.WriteFile(path.Join(testRepo.DataPath(), "root", "images", "medium", image.Filename), []byte("image"), os.ModePerm)
+		if err != nil {
+			t.Error(err)
+		}
+		err = ioutil.WriteFile(path.Join(testRepo.DataPath(), "root", "images", "small", image.Filename), []byte("image"), os.ModePerm)
+		if err != nil {
+			t.Error(err)
+		}
+		err = ioutil.WriteFile(path.Join(testRepo.DataPath(), "root", "images", "tiny", image.Filename), []byte("image"), os.ModePerm)
+		if err != nil {
+			t.Error(err)
+		}
+		err = ioutil.WriteFile(path.Join(testRepo.DataPath(), "root", "images", "original", image.Filename), []byte("image"), os.ModePerm)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+
+	checkImage := func(image *pb.Listing_Item_Image) {
+		largeID, err := cid.Decode(image.Large)
+		if err != nil {
+			t.Error(err)
+		}
+		if largeID.Version() > 0 {
+			t.Error("Large image failed to migrate to v0")
```

Comment:
These can remain as `Error` since it doesn't affect the success of the rest of the test.